test/src/test/java/hudson/console/HyperlinkNoteTest.java

```diff
@@ -72,6 +80,28 @@ public void textWithNewlinesModelHyperlinkNote() throws Exception {
                 containsString(new ModelHyperlinkNote("", 0).extraAttributes()),
                 containsString(">" + noteTextSanitized + "</a>")));
     }
+    
+    @Test
+    public void textWithSingleQuote() throws Exception {
+    	FreeStyleProject upstream = r.createFreeStyleProject("upstream");
+    	r.createFreeStyleProject("d0wnstr3'am");
+    	upstream.getPublishersList().add(new BuildTrigger("d0wnstr3'am", Result.SUCCESS));
+    	FreeStyleBuild b = r.buildAndAssertSuccess(upstream);
+        r.waitUntilNoActivity();
+        HtmlPage rsp = r.createWebClient().goTo(b.getUrl()+"console");
+        System.out.println("Before:\n"+rsp.getWebResponse().getContentAsString()+"\n\n");
+        //This would fail if job name has `'`
+        try {
+        	assertThat(String.valueOf(rsp.getAnchorByText("d0wnstr3'am").click().getWebResponse().getStatusCode()), containsString("200"));
+        } catch(Exception enfe) {
+        	String str = "str->\n";
+			str += rsp.getWebResponse().getContentAsString()+"\n\n";
+        	for(HtmlAnchor ha: rsp.getAnchors()) {
+        		str += "getNameAttribute: " + ha.getNameAttribute() + " getTextContent: " + ha.getTextContent() + "\n";
+        	}
+        	throw new Exception(str);
+        }
```

Comment:
You could assert that the list of anchors returned by `rsp.getAnchors()` contains specific items, knowing that when the assertion fails, Hamcrest will print the contents of the list for diagnosis.